dali/operators/sequence/element_extract.h

```diff
@@ -18,50 +18,88 @@
 
 #include <vector>
 #include "dali/core/common.h"
+#include "dali/core/format.h"
+#include "dali/core/tensor_layout.h"
 #include "dali/core/tensor_shape.h"
+#include "dali/kernels/common/scatter_gather.h"
 #include "dali/pipeline/operator/operator.h"
 
 namespace dali {
 
 namespace detail {
-  static void CheckInputShape(const TensorShape<>& tensor_shape,
-                              const std::vector<int>& element_map) {
-    DALI_ENFORCE(tensor_shape.size() > 0);
-    auto N_input = tensor_shape[0];
+  static void CheckInputShape(const span<const int64_t>& tensor_shape,
+                              const std::vector<int>& element_map,
+                              const TensorLayout& input_layout) {
+    if (!input_layout.empty()) {
+      DALI_ENFORCE(
+          VideoLayoutInfo::IsSequence(input_layout),
+          make_string("Input layout must describe a sequence - it must start with 'F', got '",
+                      input_layout, "' instead."));
+    }
 
-    int N_output = element_map.size();
-    DALI_ENFORCE(N_input >= N_output,
-        "Requested more elements than available");
+    DALI_ENFORCE(tensor_shape.size() > 1,
```

Comment:
Nitpick:
```suggestion
    DALI_ENFORCE(tensor_shape.dim() > 1,
```
It's preferable to use dim, since size has a different meaning for TensorShape and TensorListShape

Reply:
It's a span here. But I can maybe check the dim only once.